app/src/main/java/doit/study/droid/topic/TopicFragment.kt

```diff
@@ -16,14 +17,15 @@ import javax.inject.Inject
 class TopicFragment: Fragment(), SearchView.OnQueryTextListener {
     @Inject
     lateinit var viewModelFactory: ViewModelProvider.Factory
-    private lateinit var viewModel: TopicViewModel
+//    private lateinit var viewModel: TopicViewModel
+    private val viewModel by viewModels<TopicViewModel> { viewModelFactory }
     private lateinit var viewDataBinding: FragmentTopicBinding
 
     override fun onCreate(savedInstanceState: Bundle?) {
         App.dagger.inject(this)
         super.onCreate(savedInstanceState)
 
-        viewModel = ViewModelProviders.of(this, viewModelFactory)[TopicViewModel::class.java]
+//        viewModel = ViewModelProviders.of(this, viewModelFactory)[TopicViewModel::class.java]
```

Comment:
Don't keep commented code in the repository - it's a bad smell, because diff is becoming less readable. It's git, just remove it and you can see previous state from commits.